process/block/preprocess/sovereignChainTransactions.go

```diff
@@ -0,0 +1,159 @@
+package preprocess
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-core-go/data/transaction"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/storage/txcache"
+	"time"
+)
+
+type sovereignChainTransactions struct {
+	*transactions
+}
+
+// NewSovereignChainTransactionPreprocessor creates a new sovereign chain transaction preprocessor object
+func NewSovereignChainTransactionPreprocessor(
+	transactions *transactions,
+) (*sovereignChainTransactions, error) {
+	if transactions == nil {
+		return nil, process.ErrNilPreProcessor
+	}
+
+	sct := &sovereignChainTransactions{
+		transactions,
+	}
+
+	sct.scheduledTXContinueFunc = sct.shouldContinueProcessingScheduledTx
+
+	return sct, nil
+}
+
+// ProcessBlockTransactions processes all the transactions "from me" from the given body
+func (sct *sovereignChainTransactions) ProcessBlockTransactions(
+	header data.HeaderHandler,
+	body *block.Body,
+	haveTime func() bool,
+) (block.MiniBlockSlice, error) {
+	calculatedMiniBlocks, _, err := sct.processTxsFromMe(body, haveTime, header.GetPrevRandSeed())
+	return calculatedMiniBlocks, err
+}
+
+// CreateAndProcessMiniBlocks creates mini blocks from the selected transactions
+func (sct *sovereignChainTransactions) CreateAndProcessMiniBlocks(haveTime func() bool, randomness []byte) (block.MiniBlockSlice, error) {
+	startTime := time.Now()
+
+	//TODO: Check through "allin system tests" if this value of 2x (normal txs + scheduled txs) is enough or could be raised,
+	//as with the new mechanism of proposing and parallel processing for proposer and validators, we are not limited anymore
+	//for a subround block duration of maximum 25% from the whole round. Actually the processing time could / will go up to 60% - 80%.
+	gasBandwidth := sct.economicsFee.MaxGasLimitPerBlock(sct.shardCoordinator.SelfId()) * 2.0
```

Comment:
I think we do not need the multiplier of 2.0 here. Instead we should change the MaxGasLimitPerBlock from the TOML.

Reply:
Yep, probably we will have something like MaxGasLimitPerSovereignBlock, but first we should determine through testing how much we can push the limit.  Afterwards the TODO will be removed and we can use one variable form TOML.

Reply:
not quite another config like MaxGasLimitPerSovereignBlock because we will have another repo containing configs for the sovereign chains. There we can alter the original configs of the node the way we wanted and then they will map automatically. So yes, we can alter them whenever we want but I would not create a new set of configs

Reply:
Agree